### PR TITLE
[3.5] Security: use distroless base image to address critical Vulnerabilities

### PR DESCRIPTION
**Pull request by ahrtr**
Command:
```
trivy image --severity CRITICAL gcr.io/etcd-development/etcd:v3.5.6 
```

See report:
Screen Shot 2022-12-19 at 07 52 44

Signed-off-by: Benjamin Wang <wachao@vmware.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @mitake @ptabor @serathius @spzala

### CODE REVIEW

**Comment by ahrtr** [comment]
Note: I have confirmed that there is no critical Vulnerabilities any more after using the distroless base image.

File: Dockerfile-release.amd64 [hunk 1]
```diff
@@ -1,5 +1,8 @@
-# TODO: move to k8s.gcr.io/build-image/debian-base:bullseye-v1.y.z when patched
-FROM debian:bullseye-20220328
+FROM --platform=linux/amd64 busybox:1.34.1 as source
+FROM --platform=linux/amd64 gcr.io/distroless/base-debian11
+
+COPY --from=source /bin/sh /bin/sh
+COPY --from=source /bin/mkdir /bin/mkdir
```
Comment (fuweid) [inline Dockerfile-release.amd64:5]: is there any components using command `/bin/mkdir`?

Comment (ahrtr) [reply Dockerfile-release.amd64:5]: see line 7 and 8 in this file.

**Review by fuweid — APPROVED** [review]
LGTM(non-binding)

**Review by ptabor — APPROVED** [review]
Please mention it in a changelog. 

It's a potentially breaking change (if someone installs own scripts on top of etcd image).

**Comment by ahrtr** [comment]
> Please mention it in a changelog.

Yes, I will update the changelog when I finish all the Vulnerabilities (including  HIGH and critical)

**Comment by ahrtr** [comment]
Thanks both @ptabor and @fuweid

**Pull request merged**